core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabase.java

```diff
@@ -117,6 +117,11 @@ public Scanner scan(Scan scan) throws ExecutionException {
       close(connection);
       throw new ExecutionException(
           CoreError.JDBC_ERROR_OCCURRED_IN_SELECTION.buildMessage(e.getMessage()), e);
+    } catch (Exception e) {
+      if (connection != null) {
+        close(connection);
+      }
```

Comment:
The null handling is already done in the `close()` method, so we don't need to do that here:
```suggestion
      close(connection);
```